core/src/main/scala/org/allenai/common/Version.scala

```diff
@@ -81,19 +83,18 @@ object Version {
     val pkg = "/" + org + "/" + name.replaceAll("-", "")
     val artifactConfUrl = this.getClass.getResource(pkg + "/artifact.conf")
     val gitConfUrl = this.getClass.getResource(pkg + "/git.conf")
-
+    val cacheKeyConfUrl = this.getClass.getResource(pkg + "/cacheKey.conf")
```

Comment:
Hrm, I realized a problem with this: You're generating this as a resource file, but you're also generating it *after* you've already built the jar, so it won't get embedded.

You should be generating the cache key file within the staging directory, after the build is staged, and reading it from a system property path.

Right now, the deploy task looks something like:

  * Run in parallel:
    * stage -> generate cache key
    * clean
    * stage

which is not good. There should be only one stage step.

You should update the deploy plugin to have a task that replaces [the stage dependency](https://github.com/allenai/sbt-plugins/blob/master/src/main/scala/org/allenai/plugins/DeployPlugin.scala#L191) that does something like:
```scala
newTask := {
  val stageDir = (UniversalPlugin.autoImport.stage in thisProject).value
  // Checksum stage dir, gen cache file in `conf` or `lib` or something.
  stageDir
}
```